phoenix-core/src/main/java/org/apache/phoenix/monitoring/connqueryservice/PhoenixConnectionQueryServiceMetrics.java

```diff
@@ -0,0 +1,116 @@
+    public enum ConnectionQueryServiceMetrics {
+        CONNECTION_QUERY_SERVICE_OPEN_PHOENIX_CONNECTIONS_COUNTER(OPEN_PHOENIX_CONNECTIONS_COUNTER),
+        CONNECTION_QUERY_SERVICE_OPEN_INTERNAL_PHOENIX_CONNECTIONS_COUNTER(OPEN_INTERNAL_PHOENIX_CONNECTIONS_COUNTER),
+        CONNECTION_QUERY_SERVICE_PHOENIX_CONNECTIONS_THROTTLED_COUNTER(PHOENIX_CONNECTIONS_THROTTLED_COUNTER);
+
+        private MetricType metricType;
+        private PhoenixConnectionProfileMetric metric;
+
+        ConnectionQueryServiceMetrics(MetricType metricType) {
+            this.metricType = metricType;
+        }
+    }
+
+    private final String connectionQueryServiceName;
+    private Map<MetricType, PhoenixConnectionProfileMetric> metricRegister;
+    private ConnectionQueryServiceMetricsHistograms connectionQueryServiceMetricsHistograms;
+
+    public PhoenixConnectionQueryServiceMetrics(final String connectionQueryServiceName,
+            Configuration conf) {
+        this.connectionQueryServiceName = connectionQueryServiceName;
+        metricRegister = new HashMap<>();
+        for (ConnectionQueryServiceMetrics connectionQueryServiceMetric
+                : ConnectionQueryServiceMetrics.values()) {
+            connectionQueryServiceMetric.metric =
+                    new PhoenixConnectionProfileMetricImpl(connectionQueryServiceMetric.metricType);
+            metricRegister.put(connectionQueryServiceMetric.metricType,
+                    connectionQueryServiceMetric.metric);
+        }
+        connectionQueryServiceMetricsHistograms =
+                new ConnectionQueryServiceMetricsHistograms(connectionQueryServiceName, conf);
+    }
+
+    /**
+     * This function is used to update the value of Metric
+     * In case of counter val will be passed as 1.
+     *
+     * @param type metric type
+     * @param val update value. In case of counters, this will be 1
+     */
+    public void changeMetricValue(MetricType type, long val) {
+        if (!metricRegister.containsKey(type)) {
+            return;
+        }
+        PhoenixConnectionProfileMetric metric = metricRegister.get(type);
+        metric.set(val);
+    }
+
+    /**
+     * This function is used to get the value of Metric.
+     *
+     * @param type metric type
+     * @return val current value of metric.
+     */
+    public long getMetricValue(MetricType type) {
+        if (!metricRegister.containsKey(type)) {
+            return 0;
+        }
+        PhoenixConnectionProfileMetric metric = metricRegister.get(type);
+        return metric.getValue();
+    }
+
+    public String getConnectionQueryServiceName() {
+        return connectionQueryServiceName;
+    }
+
+    /**
+     * This method is called to aggregate all the Metrics across all Connection Query Service.
+     *
+     * @return map of Connection Query Service name -> list of ConnectionQueryServiceMetric.
+     */
+    public List<PhoenixConnectionProfileMetric> getMetricMap() {
```

Comment:
getMetrics? this method is not really returning a map right?

Reply:
Changed name to `getAllMetrics()`